lib/src/config.rs

```diff
@@ -1,37 +1,39 @@
 use bitcoin::Network;
+#[cfg(feature = "clap")]
 use clap::Parser;
 use std::path::{Path, PathBuf};
 
 /// Configuration parameters, most important the bitcoin blocks directory
-#[derive(Parser, Debug, Clone)]
+#[cfg_attr(feature = "clap", derive(Parser))]
+#[derive(Debug, Clone)]
 pub struct Config {
     /// Blocks directory (containing `blocks*.dat`)
-    #[arg(short, long)]
+    #[cfg_attr(feature = "clap", arg(short, long))]
     pub blocks_dir: PathBuf,
 
     /// Network (bitcoin, testnet, regtest, signet)
-    #[arg(short, long)]
+    #[cfg_attr(feature = "clap", arg(short, long))]
     pub network: bitcoin::Network,
 
     /// Skip calculation of previous outputs, it's faster and it uses much less memory
     /// however make it impossible calculate fees or access tx input previous scripts
-    #[arg(short, long)]
+    #[cfg_attr(feature = "clap", arg(short, long))]
     pub skip_prevout: bool,
 
     /// Maximum length of a reorg allowed, during reordering send block to the next step only
     /// if it has `max_reorg` following blocks. Higher is more conservative, while lower faster.
     /// When parsing testnet blocks, it may be necessary to increase this a lot
-    #[arg(short, long, default_value = "6")]
+    #[cfg_attr(feature = "clap", arg(short, long, default_value = "6"))]
```

Comment:
This is great to do now that the crate is separate, but ideally this should have been a move only commit, followed by changes like this one.